airflow/providers/amazon/aws/hooks/emr_containers.py

```diff
@@ -0,0 +1,217 @@
+                id=job_id,
+            )
+            reason = response['jobRun']['failureReason']
+        except KeyError:
+            self.log.error('Could not get status of the EMR on EKS job')
+        except ClientError as ex:
+            self.log.error('AWS request failed, check logs for more info: %s', ex)
+
+        return reason
+
+    def check_query_status(self, job_id: str) -> Optional[str]:
+        """
+        Fetch the status of submitted job run. Returns None or one of valid query states.
+        See: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/emr-containers.html#EMRContainers.Client.describe_job_run  # noqa: E501
+        :param job_id: Id of submitted job run
+        :type job_id: str
+        :return: str
+        """
+        try:
+            response = self.conn.describe_job_run(
+                virtualClusterId=self.virtual_cluster_id,
+                id=job_id,
+            )
+            return response["jobRun"]["state"]
+        except self.conn.exceptions.ResourceNotFoundException:
+            # If the job is not found, we raise an exception as something fatal has happened.
+            raise AirflowException(f'Job ID {job_id} not found on Virtual Cluster {self.virtual_cluster_id}')
+        except ClientError as ex:
+            # If we receive a generic ClientError, we swallow the exception so that the
+            self.log.error('AWS request failed, check logs for more info: %s', ex)
+            return None
+
+    def poll_query_status(
+        self, job_id: str, max_tries: Optional[int] = None, poll_interval: int = 30
+    ) -> Optional[str]:
+        """
+        Poll the status of submitted job run until query state reaches final state.
+        Returns one of the final states.
+        :param job_id: Id of submitted job run
+        :type job_id: str
+        :param max_tries: Number of times to poll for query state before function exits
+        :type max_tries: int
+        :param poll_interval: Time (in seconds) to wait between calls to check query status on EMR
+        :type poll_interval: int
+        :return: str
+        """
+        try_number = 1
+        final_query_state = None  # Query state when query reaches final state or max_tries reached
+
+        # TODO: Make this logic a little bit more robust.
+        # Currently this polls until the state is *not* one of the INTERMEDIATE_STATES
+        # While that should work in most cases...it might not. :)
+        while True:
+            query_state = self.check_query_status(job_id)
+            if query_state is None:
+                self.log.info(f"Try {try_number}: Invalid query state. Retrying again")
+            elif query_state in self.INTERMEDIATE_STATES:
+                self.log.info(f"Try {try_number}: Query is still in an intermediate state - {query_state}")
+            else:
+                self.log.info(f"Try {try_number}: Query execution completed. Final state is {query_state}")
```

Comment:
```suggestion
                self.log.info(f"Try %s: Query execution completed. Final state is %s", try_number, query_state)
```
Please avoid formatting the string before passing it to the logger. See: https://stackoverflow.com/questions/34619790/pylint-message-logging-format-interpolation

Reply:
I'm a little confused - was this comment intended for when I use `.format()` in some of the other code?

I dug into this a little bit and [this github comment](https://github.com/PyCQA/pylint/issues/2395#issue-348786471) linked from the stackoverflow makes it seem like f-strings and %-operator are about the same performance-wise?

Happy to change it, just wanted to make sure I'm doing the right thing.

Reply:
This is not a question of performance, but mostly a question of usability.

If you send data to the logger as text template + arguments, some services e.g. Logentries, Logstash, Stackdriver can store data in two separate columns i.e. unrrendered.  You can use it to filter the occurrence of a given log event, regardless of what the arguments are. 

This also gives a clearer result when displaying text on consoles. All arguments are bolded.
<img width="504" alt="Screenshot 2021-07-12 at 22 48 10" src="https://user-images.githubusercontent.com/12058428/125353768-442d0280-e363-11eb-8ff2-646e9bef471f.png">

Reply:
Ah ok, cool - makes perfect sense thanks for the explanation. I'll update my logging statements accordingly.